### DISCUSSION TITLE
ESP-01s constant bootloop

### DISCUSSION THREAD

**Discussion by gururise** [micropython/micropython#9301] [Hardware & Peripherals]
I've tried flashing a set of 5 ESP-01s (1MB) and every chip results in the same boot-loop:

```
ets Jan  8 2013,rst cause:4, boot mode:(3,0)

wdt reset
load 0x40100000, len 30720, room 16 
tail 0
chksum 0x90
load 0x3ffe8000, len 996, room 8 
tail 12
chksum 0x80
ho 0 tail 12 room 4
load 0x3ffe83f0, len 1080, room 12 
tail 12
chksum 0xf5
csum 0xf5
�����{��o|��d�p{lc��|;�l�o��g�d��r�l�d�l ��s�l��ll{l��{d����c��c|lb��c|����dlc���no��dn���l�d$���d`�g�#���bl�crd;l{�o�����߀�#b�lcd���
MicroPython v1.19.1 on 2022-06-18; ESP module (1M) with ESP8266
Type "help()" for more information.
>>> 
```

The REPL console responds for about 1-2 seconds, but then freezes and about 5 seconds later reboots with the same exact message.

I've flashed using both Thonny and the console (and tried every flash mode possible):
`esptool.py --port /dev/ttyUSB0 --baud 115200 write_flash --flash_size=1MB -fm dout 0x0 Downloads/esp8266-1m-20220618-v1.19.1.bin
`

This bootloop occurs in all 5 of my ESP-01s modules.  Am I doing something wrong?

**Comment by robert-hh** [comment]
It could be a power supply problem. Do you have a good stable 3.3V supply for the devices? The 3.3V output of a USB/UART bridge can typically not supply sufficient current.

**Reply by gururise** [reply]
Thanks for the reply. This is initially what I thought too; however, I am using this adapter:
![image](https://user-images.githubusercontent.com/4380009/189996159-66dd230b-435f-4dc8-93d9-7b0578c40d1f.png)

It has a 1117-3.3 regulator rated at 850mA @ 3.3v.  Unless its a fake chip, it should have more than enough Pd to power the ESP-01.  To rule out power supply issues, I will try with a separate FTDI adapter + 3.3v BENCH power supply and report back.

**Comment by davefes** [comment]
I have placed a 1000uF low-esr cap on the 3V3 output of my FTDI adapters.

**Comment by scruss** [comment]
Try clearing the flash first. There was someone on the old forum having an almost identical problem, but I can't find the post. Try using esptool.py - you can see the error messages with that more easily. It can also tell you the detected flash size
